### PR TITLE
WebSocket and Orderbook

### PR DESCRIPTION
**Pull request by mbana**
**WIP:** Would be good to get some initial feedback.

---

* add `WebSocket` with an `Orderbook` that gets updated when an
`snapshot` and `l2update` arrive.
* add `Orderbook` that holds bids and asks and allows getting the
`HighestBid` and `LowestAsk`.
* update `Message` to parse `snapshot` and `l2update` messages.
* add cmd that setups the WebSocket connection for publicly available
updates, i.e., only listen to `l2udpate` for now. The `heartbeat`
channel seems to be returning times in the future or in the past, so
ignore it for now.
* do `go build` and make test run in verbose mode in Travis.

### CODE REVIEW

**Comment by chrischris292** [comment]
+1

File: orderbook/orderbook.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import (
+	"sort"
+	"sync"
+
+	gdax "github.com/banaio/go-gdax"
+)
+
+// Orderbook represents the state of the orderbook.
+type Orderbook struct {
+	bids  map[float64]float64
```
Comment (zhengyangfeng00) [inline orderbook/orderbook.go]: Just curious, you are using a float type as the map key. Would this cause problem due to precision? For example, a L2 update message doesn't correctly remove/update an existing entry due to a tiny difference in the price.

**Comment by preichenberger** [comment]
Closing this for now.

**Comment by chrischris292** [comment]
I have a L3 orderbook that is need for some refactoring...I wrote it in a day or 2. https://github.com/chrischris292/go-gdax-orderbook.

**Pull request closed without merging**